doc/conf.py

```diff
@@ -649,14 +658,27 @@ def reset_sklearn_config(gallery_conf, fname):
     "inspect_global_variables": False,
     "remove_config_comments": True,
     "plot_gallery": "True",
-    "recommender": {"enable": True, "n_examples": 5, "min_df": 12},
+    "recommender": {"enable": True, "n_examples": 4, "min_df": 12},
     "reset_modules": ("matplotlib", "seaborn", reset_sklearn_config),
 }
 if with_jupyterlite:
     sphinx_gallery_conf["jupyterlite"] = {
         "notebook_modification_function": notebook_modification_function
     }
 
+# Secondary sidebar configuration for pages generated by sphinx-gallery
+# For index of the gallery and each nested section we hide the secondary sidebar.
+# For each example page we use the default ["page-toc", "sourcelink"] even though we
+# only need "page-toc", because we need the secondary sidebar to exist so that
+# `sphinxext/move_gallery_links.py` can work correctly, while "page-toc" alone could
+# have been empty that causes the secondary sidebar to not exist.
```

Comment:
I think this long sentence should be broken in two shorter sentences or more to be easier to understand. I think I got it but it took me several times to read it before I got it.